beacon-chain/operations/attestations/prune_expired.go

```diff
@@ -25,6 +25,27 @@ func (s *Service) pruneAttsPool() {
 	}
 }
 
+// pruneExpiredExperimental prunes attestations on every prune interval.
+func (s *Service) pruneExpiredExperimental() {
+	ticker := time.NewTicker(s.cfg.pruneInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			expirySlot, err := s.expirySlot()
+			if err != nil {
+				log.WithError(err).Error("Could not get expiry slot")
+				continue
+			}
+			numExpired := s.cfg.Cache.PruneBefore(expirySlot)
```

Comment:
Instead of checking each attestation for expiry, we pass the expiry slot to the cache. That way we don't need to hash anything.